File: rxbilling-aidl/src/main/java/com/vanniktech/rxbilling/aidl/LogcatLogger.java
package com.vanniktech.rxbilling.aidl;

import android.support.annotation.NonNull;
import android.util.Log;
import com.vanniktech.rxbilling.Logger;

final class LogcatLogger implements Logger {
  private static final String TAG = "RxBilling";

  @Override public void d(@NonNull final String log) {
    Log.d(TAG, log);
  }

  @Override public void w(@NonNull final String log) {
    Log.w(TAG, log);
  }

  @Override public void w(@NonNull final Throwable throwable) {
    Log.w(TAG, throwable);
  }

  @Override public void e(@NonNull final String log) {
    Log.e(TAG, log);
  }
}

File: rxbilling-google-play-library-v4/src/main/kotlin/com/vanniktech/rxbilling/google/play/library/v4/LogcatLogger.kt
package com.vanniktech.rxbilling.google.play.library.v4

import android.util.Log
import com.vanniktech.rxbilling.Logger

internal class LogcatLogger : Logger {
  override fun d(log: String) {
    Log.d(TAG, log)
  }

  override fun w(log: String) {
    Log.w(TAG, log)
  }

  override fun w(throwable: Throwable) {
    Log.w(TAG, throwable)
  }

  override fun e(log: String) {
    Log.e(TAG, log)
  }

  companion object {
    const val TAG = "RxBilling"
  }
}
